src/bun.js/bindings/Script.cpp

```diff
@@ -0,0 +1,181 @@
+#include "root.h"
+
+#include "Script.h"
+#include "JavaScriptCore/JSObjectInlines.h"
+#include "wtf/text/ExternalStringImpl.h"
+
+#include "JavaScriptCore/FunctionPrototype.h"
+#include "JavaScriptCore/HeapAnalyzer.h"
+
+#include "JavaScriptCore/JSDestructibleObjectHeapCellType.h"
+#include "JavaScriptCore/SlotVisitorMacros.h"
+#include "JavaScriptCore/ObjectConstructor.h"
+#include "JavaScriptCore/SubspaceInlines.h"
+#include "wtf/GetPtr.h"
+#include "wtf/PointerPreparations.h"
+#include "wtf/URL.h"
+#include "JavaScriptCore/TypedArrayInlines.h"
+#include "JavaScriptCore/PropertyNameArray.h"
+#include "Buffer.h"
+#include "GCDefferalContext.h"
+#include "Buffer.h"
+
+#include <JavaScriptCore/DOMJITAbstractHeap.h>
+#include "DOMJITIDLConvert.h"
+#include "DOMJITIDLType.h"
+#include "DOMJITIDLTypeFilter.h"
+#include "DOMJITHelpers.h"
+#include <JavaScriptCore/DFGAbstractHeap.h>
+
+namespace WebCore {
+using namespace JSC;
+
+static EncodedJSValue constructScript(JSGlobalObject* globalObject, CallFrame* callFrame, JSValue newTarget = JSValue())
+{
+    VM& vm = globalObject->vm();
+    JSValue callee = callFrame->jsCallee();
+    ArgList args(callFrame);
+    JSValue sourceArg = args.at(0);
+    String source = sourceArg.isUndefined() ? emptyString() : sourceArg.toWTFString(globalObject);
+
+    Structure* structure = Script::createStructure(vm, globalObject, Script::createPrototype(vm, globalObject));
+    Script* script = Script::create(vm, globalObject, structure, source);
```

Comment:
There's some code you should cargo cult for handling `newTarget`, which is relevant if they're using ShadowRealm or if the global object differs from the one that created it (why "lexicalGlobalObject" exists)

You can see an example of that here:

https://github.com/oven-sh/bun/blob/1af4e1dce1f4e89f2581872e390c7e68150ecf29/src/bun.js/bindings/ZigGeneratedClasses.cpp#L344-L356